HW2/P5/P5.txt

```diff
@@ -0,0 +1,86 @@
+1) Serial run yields performance of between 5 to 6 simulations per second, while parallel run yields 7-8 simulations per second. Therefore performance is better for 4 thread case, but not four times.
+
+2) Spatial decomposition surprisingly yields really good performance improvement from 7 to around 300 simulations per second. This is because a large part of calculation is done comparing whether two balls are close enough to each other to induce a collision. By decomposing space, the comparison's complexity is linear O(N) instead of a quadratic O(N^2) -  a significant increase and therefore boosting up the performance.
```

Comment:
Comparing again 1 thread vs 4 threads. 4 threads yields a performance increase of 160%.